kedify-agent/templates/agent-rbac.yaml

```diff
@@ -0,0 +1,449 @@
+{{- if .Values.agent.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kedify-leader-election
+  namespace: {{ .Release.Namespace }}
+  labels:
+    control-plane: kedify-agent
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kedify-leader-election
+  namespace: {{ .Release.Namespace }}
+  labels:
+    control-plane: kedify-agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kedify-leader-election
+subjects:
+- kind: ServiceAccount
+  name: kedify-agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kedify-manager-role
+  namespace: {{ .Release.Namespace }}
+rules:
+
+# Kedify agent needs to be able to update its own container image
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - patch
+
+#  Cruding own CRD that holds the helm chart values for Keda installation, but also enabling disabling telemetry
+- apiGroups:
+  - install.kedify.io
+  resources:
+  - KedifyConfiguration
+  verbs:
+  - "*"
+
+# Save/load state of each installed resource and its version
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - create
+  - delete
+  - get
+  - patch
+  - update
+  resourceNames:
+  - kedify-state
+  - kedify-http-addon-state
```

Comment:
also luckily something we should be able to hide behind a flag, these two are used in helm deployments of KEDA and addon, neither will be performed